test/src/components/serviceInsights/serviceInsightsGraph/nodes.spec.jsx

```diff
@@ -67,6 +78,13 @@ describe('<Nodes/>', () => {
         expect(wrapper.find('circle')).to.have.lengthOf(0);
     });
 
+    it('should build map of node element references based on `id` property', () => {
+        const nodes = [{x: 10, y: 20, data: {id: 'foo'}}];
+        const wrapper = mount(<Nodes nodes={nodes} onHover={() => {}} onLeave={() => {}} />);
+        const instance = wrapper.instance();
+        expect(Object.keys(instance.nodeRefs)[0]).to.equal('foo');
```

Comment:
You can get at the instance with `shallow`, too. Was there a different reason why mount was necessary here?

Reply:
When using `shallow` it doesn't seem like Enzyme triggers the `ref()` callbacks necessary to build the `nodeRefs` state.  `instance.nodeRefs` ends up being undefined.